includes/admin/class-amp-post-meta-box.php

```diff
@@ -86,6 +86,24 @@ class AMP_Post_Meta_Box {
 	 */
 	const REST_ATTRIBUTE_NAME = 'amp_enabled';
 
+	/**
+	 * Blocks that are only meant for AMP.
+	 *
+	 * @since 2.1
+	 * @var array
+	 */
+	const AMP_DEPENDENT_BLOCKS = [
+		'amp/amp-brid-player',
+		'amp/amp-ima-video',
+		'amp/amp-jwplayer',
+		'amp/amp-mathml',
+		'amp/amp-o2-player',
+		'amp/amp-ooyala-player',
+		'amp/amp-reach-player',
+		'amp/amp-springboard-player',
+		'amp/amp-timeago',
+	];
+
```

Comment:
@pierlon On second look, I'm questioning whether it's necessary to move the hardcoded block names into PHP and handle `ampBlocks` and `ampBlocksInUse` here. `ampBlocksInUse` could be achieved in the editor, maybe by applying `withSelect` to the new higher-order component and filtering `select( 'core/block-editor' ).getBlocks()`. This should be fine performancewise because I imagine it's rare to have many AMP blocks on a page, and `getBlocks` [is memoized](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/store/selectors.js#L199-L211). 

With future deprecation in mind, I think it might make more sense to focus on refactoring on the JS side to make these blocks easily to delete from the code base eventually. I'd break the blocks into a module of their own with a single entry point in the editor JS (or put them in their own JS file, but that might be going farther than needed). Do you think this would work? Then we wouldn't have to make the PHP side of the plugin aware of these blocks that we are deprecating.